### PR TITLE
Active admin implementation

### PR DESCRIPTION
**Pull request by fedelcar**
# Summary

Active Admin implementation. 
Go to /admin and login with admin@example.com, password: password.
http://activeadmin.info/docs/documentation.html
# Trello Card

https://trello.com/c/ZSN0LrSM

### CODE REVIEW

**Comment by fedelcar** [comment]
@epintos

File: app/admin/article.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+ActiveAdmin.register Article do
+
```
Comment (epintos) [inline app/admin/article.rb:2]: Remember not to leave commented code. You can always get it back from git if you need it in the future

File: app/admin/dashboard.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+  content title: proc{ I18n.t("active_admin.dashboard") } do
+    div class: "blank_slate_container", id: "dashboard_default_message" do
+      span class: "blank_slate" do
+        span I18n.t("active_admin.dashboard_welcome.welcome")
+        small I18n.t("active_admin.dashboard_welcome.call_to_action")
+      end
+    end
+
+    # Here is an example of a simple dashboard with columns and panels.
```
Comment (epintos) [inline app/admin/dashboard.rb]: Same thing

File: app/controllers/application_controller.rb [hunk 3]
```diff
@@ -2,4 +2,10 @@ class ApplicationController < ActionController::Base
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
+  before_action :set_locale
+	before_action :authenticate_user!
+	 
+  def set_locale
```
Comment (epintos) [inline app/controllers/application_controller.rb:8]: I think you didn't add the dropdown to change the language right?

File: app/controllers/articles_controller.rb [hunk 4]
```diff
@@ -15,6 +11,9 @@ def new
     @article = Article.new
   end
 
+  def send_last_10
+    redirect_to "/"
```
Comment (epintos) [inline app/controllers/articles_controller.rb]: redirect_to root_path its better, don't you think?

File: app/controllers/articles_controller.rb [hunk 5]
```diff
@@ -15,6 +11,9 @@ def new
...
+  end
   def edit
     @article = Article.find(params[:id])
```
Comment (epintos) [inline app/controllers/articles_controller.rb]: here could be

``` ruby
def edit
   article
end
```

File: app/controllers/callbacks_controller.rb [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+        sign_in_and_redirect @user, :event => :authentication
+      else
+        session["devise.google_data"] = request.env["omniauth.auth"]
+        redirect_to new_user_registration_url
+      end
+  	end
+
+  	def self.find_for_google_oauth2(access_token, signed_in_resource=nil)
+	    data = access_token.info
+	    user = User.where(:email => data["email"]).first
```
Comment (epintos) [inline app/controllers/callbacks_controller.rb]: Its rubocop running? This should be sth detected as sth to improve

If you know there is only one you can do this:

``` ruby
User.find_by_email:(data['email'])
```

File: app/controllers/comments_controller.rb [hunk 7]
```diff
@@ -2,18 +2,22 @@ class CommentsController < ApplicationController
   http_basic_authenticate_with name: 'dhh', password: 'secret', only: :destroy
 
   def create
-    @article = Article.find(params[:article_id])
+    @article = article
```
Comment (epintos) [inline app/controllers/comments_controller.rb]: Here you can call `article` directly

File: app/models/user.rb [hunk 8]
```diff
@@ -2,5 +2,30 @@ class User < ActiveRecord::Base
...
+      user.provider = auth.provider
+      user.uid = auth.uid
+      user.email = auth.info.email
+      user.password = Devise.friendly_token[0,20]
+    end
+  end
+  
+  def self.find_for_google_oauth2(access_token, signed_in_resource=nil)
+    data = access_token.info
+    user = User.where(:email => data["email"]).first
```
Comment (epintos) [inline app/models/user.rb]: Same thing

**Comment by epintos** [comment]
Well done @fedelcar. Just a few comments

**Comment by epintos** [comment]
You can close the others PR that are deprecated

**Pull request closed without merging**